monai/apps/auto3d/data_analyzer.py

```diff
@@ -0,0 +1,541 @@
+                                get_case_summary is called to combine results and save_yaml is called to save results.
+
+    Custimize statistics calculation:
+        Write a custome function for indivisual case:
+            _get_your_stats(self):
+                1. processed_data = retrive processed data from self.data, if not exist, process data (like cropping)
+                2. define a dict case_stats = {'stats1': calculate(processed_data)}. Notice the data may be a list of
+                data from different modalities. So calculate(processed_data) should return a list of stats. case_status
+                will be written to the yaml file
+                3. create a dict case_stats_summary = {'stats1': summary_function}. The summary_function will process
+                a list of ['stats1','stats1',...] from all cases in self.dataset.datalist
+                4. update self.data with processed_data and update self.gather_summary to register case_stats_summary,
+                and return case_stats
+        Add _get_your_stats to self._hardcode_functions
+    """
+
+    def __init__(
+        self,
+        datalist: Union[str, Dict],
+        dataroot: str,
+        output_yaml: str = "./data_stats.yaml",
+        average: bool = False,
+        do_ccp: bool = True,
+        device: str = "cuda",
+        worker: int = 2,
+    ):
+        self.output_yaml = output_yaml
+        files, _ = datafold_read(datalist=datalist, basedir=dataroot, fold=-1)
+        ds = data.Dataset(
+            data=files,
+            transform=transforms.Compose(
+                [
+                    transforms.LoadImaged(keys=["image", "label"]),
+                    transforms.EnsureChannelFirstd(keys=["image", "label"]),  # this creates label to be (1,H,W,D)
+                    transforms.Orientationd(keys=["image", "label"], axcodes="RAS"),
+                    transforms.EnsureTyped(keys=["image", "label"], data_type="tensor"),
+                    # some dataset has onehot label size of (H,W,D,C). Only allow label index of size (1,H,W,D)
+                    transforms.Lambdad(
+                        keys="label", func=lambda x: torch.argmax(x, dim=0, keepdim=True) if x.shape[0] > 1 else x
+                    ),
+                    transforms.SqueezeDimd(keys=["label"], dim=0),  # make label (H,W,D)
+                ]
+            ),
+        )
+        self.dataset = data.DataLoader(ds, batch_size=1, shuffle=False, num_workers=worker, collate_fn=lambda x: x)
+        # If average all the modalities in the summary, if False, the stats for each modality are
+        # calculated separately
+        self.SUMMERY_AVERAGE = average
+        self.DO_CONNECTED_COMP = do_ccp
+        self.device = device
+        # gather all summary function for combining case stats
+        self.gather_summary: Dict[str, Dict] = {}
+        self._hardcode_functions()
+        self._hardcode_operations()
+        print(self.gather_summary)
```

Comment:
I would suggest not having print statements in code but always to log to a named logger users can attach their own printing handlers to. In testing for example code should be silent unless failure occurs.

Reply:
Thanks for the suggestions!

One of the target user group of this package is non-expert who wants to use leverage monai in their work without knowing the details. For these users, they may be less familiar with the usage of monai log. Should we print a hint/help info in the beginning?

Reply:
On a second thought, this print can be removed because the stats functions aren't really called until get_all_case_stats() is called. So it prints nothing and can be removed.